pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -116,8 +116,8 @@ def gen_program(self):
             self.gen_program_end(hart)
             for hart in range(cfg.num_of_harts):
                 self.gen_data_page_begin(hart)
-                if(cfg.no_data_page):
-                    self.gen_data_page()
+                if not cfg.no_data_page:
```

Comment:
I think this change is already in #694?

Reply:
Yes, this PR is actually just a superset of the #694, as it's needed for functionality of the code.

Reply:
ah got it, please make sure to rebase once #694 goes in

Reply:
Sure, will do.